infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -168,10 +168,6 @@ private long initStreamLogDirectory() {
             String corfuDir = logDir.getParent().toString();
             FileStore corfuDirBackend = Files.getFileStore(Paths.get(corfuDir));
 
-            if (corfuDirBackend.isReadOnly()) {
-                throw new LogUnitException("Cannot start Corfu on a read-only filesystem:" + corfuDir);
-            }
-
             File corfuDirFile = new File(corfuDir);
             if (!corfuDirFile.canWrite()) {
```

Comment:
Just saw this comment in JDK source code:
https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/io/File.java#L780

Looks like the parent filesystem check may still be necessary  in some cases in addition to the directory check here, though I understand it's still not 100% guarantee.
Can we try to increase JVM privilege to overcome this issue? Or use another directory check API with stronger guarantee?

Reply:
From the Java doc..
On some platforms it may be possible to start the Java virtual machine with special privileges that allow it to modify files that are marked read-only.
We are only interested in capability to write to a file. If the permissions allow modification of the file, that is going to be good enough for LogUnitServer.
Besides this is a small best effort check. It would be quite difficult to cover all the possibilities of file based permissions mismatch that can occur.